jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -151,25 +272,38 @@ def process(self, input):
             readnoise_model.close()
             gain_model.close()
 
-        # Save the OLS optional fit product, if it exists
-        if opt_model is not None:
-            self.save_model(opt_model, 'fitopt', output_file=self.opt_name)
-
-        # Save the GLS optional fit product, if it exists
+        # Save the GLS optional fit product, if it exists (NOT USED RIGHT NOW)
+        # When GLS is implemented, this will not be a data model
         if gls_opt_model is not None:
             self.save_model(
                 gls_opt_model, 'fitoptgls', output_file=self.opt_name
             )
 
-        if out_model is not None:
+        # Save the OLS optional fit product, if it exists
+        if opt_info is not None:
+            opt_model = create_optional_results_model(opt_info)
+            self.save_model(opt_model, 'fitopt', output_file=self.opt_name)
+
+        if image_info is not None:
+            input_model.int_times = old_int_times
+            out_model = create_output_image_model(input_model, image_info)
+            if out_model is not None:
```

Comment:
I don't see how out_model could be None here.

Reply:
Removed check.